torchaudio/prototype/functional/functional.py

```diff
@@ -0,0 +1,42 @@
+import math
+
+import torch
+
+
+def convolve(x: torch.Tensor, y: torch.Tensor, use_fft: bool = False) -> torch.Tensor:
+    """
+    Convolves inputs along last dimension. Note that, in contrast to ``torch.nn.functional.conv1d``, this
+    function applies the true `convolution`_ operator.
+
+    Args:
+        x (torch.Tensor): First convolution operand, with shape `(*, N)`.
+        y (torch.Tensor): Second convolution operand, with shape `(*, M)`
+            (leading dimensions must match those of ``x``).
+        use_fft (bool, optional): If ``True``, then convolve ``x`` and ``y`` using FFT.
+            Otherwise, convolve ``x`` and ``y`` directly via sums. (Default: ``False``)
+
+    Returns:
+        torch.Tensor: Result of convolving ``x`` and ``y``, with shape `(*, N + M - 1)`, where
+        the leading dimensions match those of ``x``.
+
+    .. _convolution:
+        https://en.wikipedia.org/wiki/Convolution
+    """
```

Comment:
the leading dimensions of `x` and `y` need to be checked before applying convolution.